### PR TITLE
Read only graph support

### PR DESCRIPTION
**Pull request by fppt**

### CODE REVIEW

**Comment by fppt** [comment]
please retest this please - ai.grakn.test.engine.GraknEngineServerTest.whenEnginePropertiesDoesNotIndicateZookeeperStorage_GraphStorageIsUsed

**Comment by fppt** [comment]
please retest this please - ai.grakn.test.engine.GraknEngineServerTest.whenEnginePropertiesDoesNotIndicateZookeeperStorage_GraphStorageIsUsed

File: grakn-engine/src/main/java/ai/grakn/engine/controller/VisualiserController.java [hunk 1]
```diff
@@ -109,7 +110,7 @@ public VisualiserController(Service spark) {
     private String conceptById(Request req, Response res) {
         String keyspace = getKeyspace(req);
 
-        try (GraknGraph graph = getInstance().getGraph(keyspace)) {
+        try (GraknGraph graph = getInstance().getGraph(keyspace, GraknTxType.WRITE)) {
```
Comment (aelred) [inline grakn-engine/src/main/java/ai/grakn/engine/controller/VisualiserController.java]: This one (and others) can probably be read-only.  This may be something for another PR though.

File: grakn-core/src/main/java/ai/grakn/GraknGraph.java [hunk 2]
```diff
@@ -382,7 +390,10 @@
     /**
      * Commits any changes to the graph and closes the transaction. You must use the {@link GraknSession} to
      * get a new open transaction.
+     *
+     * @throws GraknValidationException when the transaction contains graph mutations which does not form to the Grakn
```
Comment (fppt) [inline grakn-core/src/main/java/ai/grakn/GraknGraph.java]: correct. Typo

File: grakn-engine/src/main/java/ai/grakn/engine/controller/VisualiserController.java (hunk 1 above)
Comment (fppt) [inline grakn-engine/src/main/java/ai/grakn/engine/controller/VisualiserController.java]: Another PR. Right now there is not performance benefit of read only graphs
